### PR TITLE
Build node on Apple M1

### LINKED ISSUE #3404: Support Apple M1
Max OS BigSur , CPU Apple M1 Arm64

I made some attempt to fix the compilation isssue in apple M1.
https://github.com/wangjia184/rchain/commit/753e54c68cce7062d4b2e7da7cf0158a5be276ad

Generally `protoc` is not shipped with a pre-compiled binary, hence it has to attach to an existing `protoc` , details can be found here https://github.com/scalapb/ScalaPB/issues/1024

Still the compilation fails at [this line](https://github.com/wangjia184/rchain/blob/753e54c68cce7062d4b2e7da7cf0158a5be276ad/project/StacksafeScalapbGenerator.scala#L35): 
```
--scala_0_out: java.lang.NoSuchMethodError: 'com.google.protobuf.Descriptors$FileDescriptor com.google.protobuf.Descriptors$FileDescriptor.internalBuildGeneratedFileFrom(java.lang.String[], com.google.protobuf.Descriptors$FileDescriptor[])'
	at scalapb.options.Scalapb.<clinit>(Scalapb.java:22457)
	at scalapb.options.compiler.Scalapb$.registerAllExtensions(Scalapb.scala:8)
	at coop.rchain.scalapb.StacksafeScalapbGenerator$.registerExtensions(StacksafeScalapbGenerator.scala:35)
	at protocgen.CodeGenApp.run(CodeGenApp.scala:42)
	at protocgen.CodeGenApp.run$(CodeGenApp.scala:39)
  | => mat coop.rchain.scalapb.StacksafeScalapbGenerator$.run(StacksafeScalapbGenerator.scala:33)
  | => sat protocgen.CodeGenApp.run(CodeGenApp.scala:31)
  | => rat protocgen.CodeGenApp.run$(CodeGenApp.scala:30)
  | => mat coop.rchain.scalapb.StacksafeScalapbGenerator$.run(StacksafeScalapbGenerator.scala:33)
  | => cat protocbridge.frontend.PluginFrontend$.$anonfun$runWithBytes$1(PluginFrontend.scala:52)
	at scala.util.Try$.apply(Try.scala:213)
	at protocbridge.frontend.PluginFrontend$.runWithBytes(PluginFrontend.scala:52)
	at protocbridge.frontend.PluginFrontend$.runWithInputStream(PluginFrontend.scala:122)
	at protocbridge.frontend.PosixPluginFrontend$.$anonfun$prepare$2(PosixPluginFrontend.scala:39)
	at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:23)
	at scala.concurrent.impl.ExecutionContextImpl$DefaultThreadFactory$$anon$1$$anon$2.block(ExecutionContextImpl.scala:75)
	at java.base/java.util.concurrent.ForkJoinPool.managedBlock(ForkJoinPool.java:3127)
	at scala.concurrent.impl.ExecutionContextImpl$DefaultThreadFactory$$anon$1.blockOn(ExecutionContextImpl.scala:87)
	at scala.concurrent.package$.blocking(package.scala:146)
	at protocbridge.frontend.PosixPluginFrontend$.$anonfun$prepare$1(PosixPluginFrontend.scala:37)
	at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:23)
	at scala.concurrent.Future$.$anonfun$apply$1(Future.scala:659)
	at scala.util.Success.$anonfun$map$1(Try.scala:255)
	at scala.util.Success.map(Try.scala:213)
	at scala.concurrent.Future.$anonfun$map$1(Future.scala:292)
	at scala.concurrent.impl.Promise.liftedTree1$1(Promise.scala:33)
	at scala.concurrent.impl.Promise.$anonfun$transform$1(Promise.scala:33)
	at scala.concurrent.impl.CallbackRunnable.run(Promise.scala:64)
	at java.base/java.util.concurrent.ForkJoinTask$RunnableExecuteAction.exec(ForkJoinTask.java:1434)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:295)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1016)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1665)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1598)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:183)
```

I am appreciated if the project can support M1 CPU.  Currently I am using a UTM VirtualMachine to build thee project, that is very slow 

Thanks in advancced

### PR DESCRIPTION
**Pull request by nzpr**
## Overview
This PR updates sbt-protoc plugin, latest version uses protoc version that is compatible with M1. Closes https://github.com/rchain/rchain/issues/3404

The second commits is not really necessary, but this warning occurred in sbt shell on M1 so sbt is also updated and deprecated syntax is adjusted.
```
Exception in thread "sbt-socket-server" java.lang.UnsatisfiedLinkError: Can't load library: /Users/nzpr/Library/Caches/JNA/temp/jna8192809878255910661.tmp
	at java.base/java.lang.ClassLoader.loadLibrary(ClassLoader.java:2633)
	at java.base/java.lang.Runtime.load0(Runtime.java:768)
	at java.base/java.lang.System.load(System.java:1837)
	at com.sun.jna.Native.loadNativeDispatchLibraryFromClasspath(Native.java:1018)
	at com.sun.jna.Native.loadNativeDispatchLibrary(Native.java:988)
	at com.sun.jna.Native.<clinit>(Native.java:195)
	at com.sun.jna.Structure.setAlignType(Structure.java:280)
	at com.sun.jna.Structure.<init>(Structure.java:197)
	at com.sun.jna.Structure.<init>(Structure.java:193)
	at com.sun.jna.Structure.<init>(Structure.java:180)
	at com.sun.jna.Structure.<init>(Structure.java:172)
	at org.scalasbt.ipcsocket.UnixDomainSocketLibrary$SockaddrUn.<init>(UnixDomainSocketLibrary.java:83)
	at sbt.internal.server.Server$$anon$1$$anon$2.$anonfun$run$1(Server.scala:65)
	at scala.util.Try$.apply(Try.scala:213)
	at sbt.internal.server.Server$$anon$1$$anon$2.run(Server.scala:60)
```

### Please make sure that this PR:
- [x] is at most 200 lines of code (excluding tests),
- [x] meets [RChain development coding standards](https://rchain.atlassian.net/wiki/spaces/DOC/pages/28082177/Coding+Standards),
- [x] includes tests for all added features,
- [x] has a reviewer assigned,
- [x] has [all commits signed](https://rchain.atlassian.net/wiki/spaces/DOC/pages/498630673/How+to+sign+commits+to+rchain+rchain).

### [Bors](https://bors.tech/) cheat-sheet:

- `bors r+` runs integration tests and merges the PR (if it's approved),
- `bors try` runs integration tests for the PR,
- `bors delegate+` enables non-maintainer PR authors to run the above.

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -15,13 +15,12 @@ lazy val projectSettings = Seq(
...
-    Resolver.sonatypeRepo("snapshots"),
-    "jitpack" at "https://jitpack.io"
-  ),
+  resolvers ++=
+    Resolver.sonatypeOssRepos("releases") ++
+    Resolver.sonatypeOssRepos("snapshots") ++
+    Seq("jitpack" at "https://jitpack.io"),
   wartremoverExcluded += sourceManaged.value,
-  wartremoverErrors in (Compile, compile) ++= Warts.allBut(
+  Compile / compile / wartremoverErrors ++= Warts.allBut(
```
Comment (tgrospic) [inline build.sbt:23]: Kudos for using the new sbt syntax.

**Review by tgrospic — APPROVED** [review]
LGTM

**Review by stanislavlyalin — APPROVED** [review]

**Comment by nzpr** [comment]
bors merge

**Comment by nzpr** [comment]
bors merge

**Pull request merged**